### ISSUE TITLE
msgpack-python failing in ubuntu 12.04 docker container

### ISSUE THREAD

**Issue by karlgrz** [saltstack/salt-bootstrap#229]
Seeing the following output multiple runs, using the development and stable bootstrap:

```
Reading package lists...
Reading package lists...
Building dependency tree...
Reading state information...
Package msgpack-python is not available, but is referred to by another package.
This may mean that the package is missing, has been obsoleted, or
is only available from another source

E: Package 'msgpack-python' has no installation candidate
 * ERROR: Failed to run install_ubuntu_git_deps()!!!
```

This is a Ubuntu 12.04 docker container. I'd be happy to provide any more details necessary.

**Comment by s0undt3ch** [comment]
Can you please confirm if you have the universe repository enabled?

**Comment by s0undt3ch** [comment]
``` diff
diff --git a/bootstrap-salt.sh b/bootstrap-salt.sh
index b121122..c792588 100755
--- a/bootstrap-salt.sh
+++ b/bootstrap-salt.sh
@@ -1115,10 +1115,19 @@ install_ubuntu_deps() {
     else
         __apt_get_noinput python-software-properties || return 1
     fi
+
+    if [ "x$(grep -R universe /etc/apt/sources.list /etc/apt/sources.list.d/ | grep -v '#')" != "x" ]; then
+        __IS_UNIVERSE_ENABLED=$BS_TRUE
+    else
+        __IS_UNIVERSE_ENABLED=$BS_FALSE
+    fi
+
     if [ $DISTRO_MAJOR_VERSION -lt 11 ] && [ $DISTRO_MINOR_VERSION -lt 10 ]; then
         add-apt-repository ppa:saltstack/salt || return 1
+        add-apt-repository "deb http://archive.ubuntu.com/ubuntu $(lsb_release -sc) universe" || return 1
     else
         add-apt-repository -y ppa:saltstack/salt || return 1
+        add-apt-repository -y "deb http://archive.ubuntu.com/ubuntu $(lsb_release -sc) universe" || return 1
     fi

     apt-get update
```

Can you please update your salt-bootstrap script to include the patch above and retry?

**Comment by karlgrz** [comment]
Of course I don't have the universe repository enabled, why would I do such a silly thing?! :-)

Thank you for the help, closing!

**Comment by s0undt3ch** [comment]
Are you explicitly disabling it? No right? So the script should enable it for you.

**Comment by karlgrz** [comment]
The base image I was using for the docker container only had main repository in /etc/apt/sources.list. After adding that repo and running apt-get update the bootstrap worked just fine.

**Comment by s0undt3ch** [comment]
Could you please try my diff above on a clean image? That would save me a lot of work :smile:

**Comment by karlgrz** [comment]
Certainly! Hold please...

**Comment by karlgrz** [comment]
Script on a clean image did not run successfully. Still get the same error message.

**Comment by s0undt3ch** [comment]
With the changes I pasted above?

**Comment by karlgrz** [comment]
Yes. 

What I did was:
- boot ubuntu base docker image
- apt-get install -y vim
- apt-get install -y curl
- curl https://raw.github.com/saltstack/salt-bootstrap/stable/bootstrap-salt.sh -o bootstrap-salt.sh
- manually applied your patch from above
- chmod 755 bootstrap-salt.sh
- (as root) ./bootstrap-salt.sh git develop

**Comment by s0undt3ch** [comment]
Bummer!

Ok, Thank You. I need to get docker a test run.

**Comment by karlgrz** [comment]
No problem. If it helps at all, when I look at /etc/apt/sources.list after running the script with your changes the universe repo is not listed, just main. It's almost like that add-apt-repository command never got executed.

**Comment by s0undt3ch** [comment]
it should be under /etc/apt/sources.list.d/ in a separate file

**Comment by karlgrz** [comment]
Hey, I went back and tried this again from a fresh image. This time I forked the repo and updated the script in that, and just referenced the entire repo in my docker container. 

When I ran the script this way everything worked appropriately. I think my first attempt might have been marred by a typo or something else. 

I repeated this twice, so I think it would be good to go. Thanks again for the help!
